content/oci/storage.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package oci
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/errdef"
+	"oras.land/oras-go/v2/internal/ioutil"
+)
+
+// Storage is a CAS based on file system with the OCI-Image layout.
+// Reference: https://github.com/opencontainers/image-spec/blob/master/image-layout.md
+type Storage struct {
+	// blobRoot is the root directory of the stored blobs.
+	blobRoot string
+	// ingestRoot is the root directory of the temporary ingest files.
+	ingestRoot string
+}
+
+func NewStorage(root string) *Storage {
+	return &Storage{
+		blobRoot:   filepath.Join(root, "blobs"),
+		ingestRoot: filepath.Join(root, "ingest"),
+	}
+}
+
+// Fetch fetches the content identified by the descriptor.
+func (s *Storage) Fetch(_ context.Context, target ocispec.Descriptor) (io.ReadCloser, error) {

```

Comment:
Just curious why context parameter is ignored in Storage implementation? What is it used for?

Reply:
This parameter is defined in the methods of the [Storage interfaces](https://github.com/oras-project/oras-go/blob/main/content/storage.go#L25-L38). The OCI storage does not use context, but other targets may use it.